qiskit_experiments/calibration_management/backend_calibrations.py

```diff
@@ -429,3 +438,62 @@ def operated_qubits(self) -> Dict[int, List[int]]:
                 self._operated_qubits[len(coupling)].append(coupling)
 
         return self._operated_qubits
+
+    @classmethod
+    def from_exp_data(
+        cls,
+        experiment_data: ExperimentData,
+        backend: Backend
+    ) -> Optional["BackendCalibrations"]:
+        """Return backend calibrations extracted from experiment data.
+
+        The calibrations are only built if they were created from a library.
+
+        Args:
+            experiment_data: In the metadata of the experiment data there may
+                be calibration data which tells us how to build the calibrations.
+            backend: The backend for which to build the calibrations.
+        """
+
+        cal_metadata = experiment_data.metadata.get("calibrations", None)
+        if cal_metadata is None:
+            warn(f"No calibration metadata in metadata. Returning None.")
+            return None
+
+        # Create the library
+        lib_name = cal_metadata.get("library", None)
+        if lib_name is None:
+            warn(f"Cannot load {cls.__name__} without a library. Returning None.")
+            return None
+
+        lib_class = getattr(sys.modules["qiskit_experiments"].calibration_management, lib_name)
+        basis_gates = cal_metadata.get("basis gates", None)
+        default_values = cal_metadata.get("default values", None)
+
+        library = lib_class(basis_gates=basis_gates, default_values=default_values)
+
+        # Create the calibrations
+        backend_name = cal_metadata.get("backend name", None)
+        if backend_name != backend.name():
+            raise CalibrationError(
+                f"The name of the given backend {backend.name()} does not match the name "
+                f"in the calibration metadata {backend_name}.")
+
+        cals = BackendCalibrations(backend, library=library)
+
+        # Populate the calibrations with the parameter values in the metadata
+        param_values = cal_metadata.get("calibration parameters", [])
+        for val in param_values:
+
+            param_value = ParameterValue(
+                **{key: val for key, val in val.items() if key in ParameterValue.__annotations__}
```

Comment:
`[f.name for f in dataclasses.fields(ParameterValue)]` might be a more proper place to check.

Reply:
Something like this? d8ec759